sql/moz-fx-data-shared-prod/firefox_ios_derived/firefox_ios_clients_v1/query.sql

```diff
@@ -0,0 +1,245 @@
+    NULLIF(metrics.string.adjust_network, "") AS adjust_network,
+  FROM
+    firefox_ios.first_session
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+    AND client_info.client_id IS NOT NULL
+),
+first_session_ping AS (
+  SELECT
+    client_id,
+    MIN(sample_id) AS sample_id,
+    -- DATETIME(MIN(submission_timestamp)) AS min_submission_datetime,
+    -- DATE(MIN(SAFE.PARSE_DATETIME('%F', SUBSTR(first_run_date, 1, 10)))) AS first_run_date,
+    ARRAY_AGG(
+      IF(
+        -- we should probably do the nullif prior to doing this is not null check
+        adjust_ad_group IS NOT NULL
+        OR adjust_campaign IS NOT NULL
+        OR adjust_creative IS NOT NULL
+        OR adjust_network IS NOT NULL,
+        STRUCT(
+          submission_timestamp,
+          adjust_ad_group,
+          adjust_campaign,
+          adjust_creative,
+          adjust_network
+        ),
+        NULL
+      ) IGNORE NULLS
+      ORDER BY
+        submission_timestamp
+      LIMIT
+        1
+    )[SAFE_OFFSET(0)] AS adjust_info,
+  FROM
+    first_session_ping_base
+  GROUP BY
+    client_id
+),
+-- Find earliest data per client from the metrics ping.
+metrics_ping_base AS (
+  SELECT
+    client_info.client_id AS client_id,
+    sample_id,
+    submission_timestamp,
+    NULLIF(fxa_metrics.metrics.string.adjust_ad_group, "") AS adjust_ad_group,
+    NULLIF(fxa_metrics.metrics.string.adjust_campaign, "") AS adjust_campaign,
+    NULLIF(fxa_metrics.metrics.string.adjust_creative, "") AS adjust_creative,
+    NULLIF(fxa_metrics.metrics.string.adjust_network, "") AS adjust_network,
+    -- NULLIF(metrics.string.metrics_install_source, "") AS install_source,
+  FROM
+    firefox_ios.metrics AS fxa_metrics
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+    AND client_info.client_id IS NOT NULL
+),
+metrics_ping AS (
+  SELECT
+    client_id,
+    MIN(sample_id) AS sample_id,
+    -- DATETIME(
+    --   MIN(submission_timestamp)
+    -- ) AS min_submission_datetime,  -- do we need this? Looks like later we might want to see timestamp corresponding to the row that contains the campaign info
+    ARRAY_AGG(
+      IF(
+        adjust_ad_group IS NOT NULL
+        OR adjust_campaign IS NOT NULL
+        OR adjust_creative IS NOT NULL
+        OR adjust_network IS NOT NULL,
+        STRUCT(
+          submission_timestamp,
+          adjust_ad_group,
+          adjust_campaign,
+          adjust_creative,
+          adjust_network
+        ),
+        NULL
+      ) IGNORE NULLS
+      ORDER BY
+        submission_timestamp
+      LIMIT
+        1
+    )[SAFE_OFFSET(0)] AS adjust_info,
+  FROM
+    metrics_ping_base
+  GROUP BY
+    client_id
+),
+_current AS (
+  SELECT
+    * EXCEPT (sample_id, adjust_info, activated),
+    COALESCE(first_session.adjust_info, metrics.adjust_info) AS adjust_info,
+    COALESCE(first_seen.sample_id, first_session.sample_id, metrics.sample_id) AS sample_id,
```

Comment:
We should instead include `sample_id` in GROUP BYs and JOINs, rather than aggregating

Reply:
Is it possible for `sample_id` to change for a client?

Reply:
No, it will never change. You can see how it's implemented here: https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/udf_js/sample_id/udf.sql